enf/domain_ratelimit.go

```diff
@@ -0,0 +1,66 @@
+package enf
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+// DomainService handles communication with the domain-related methods
+// of the ENF API.
+type DomainService service
+
+// DomainRateLimits represents the values of rate limits for a domain.
+type DomainRateLimits struct {
+	PacketsPerSecond *int `json:"packets_per_second"`
+	PacketsBurstSize *int `json:"packets_burst_size"`
+	BytesPerSecond   *int `json:"bytes_per_second"`
+	BytesBurstSize   *int `json:"bytes_burst_size"`
+}
+
+// domainRateLimitResponse represents the standard API response for
+// all relevant network or endpoint rate limit endpoints.
+type domainRateLimitResponse struct {
+	Data []*DomainRateLimits    `json:"data"`
+	Page map[string]interface{} `json:"page"`
+}
+
+// GetDefaultRateLimits gets the default rate limits for the given domain.
+func (s *DomainService) GetDefaultRateLimits(ctx context.Context, domain string) (*DomainRateLimits, *http.Response, error) {
+	endpoint := fmt.Sprintf("api/xcr/v2/domains/%v/ep_rate_limits/default", domain)
+	body, resp, err := s.client.get(ctx, endpoint, new(domainRateLimitResponse))
+	if err != nil {
+		return nil, resp, err
+	}
+	return body.(*domainRateLimitResponse).Data[0], resp, nil
+}
+
+// GetMaxRateLimits gets the rate limits for the given domain.
+func (s *DomainService) GetMaxRateLimits(ctx context.Context, domain string) (*DomainRateLimits, *http.Response, error) {
```

Comment:
Rather than `GetMaxRateLimits`, let's do `GetMaxDefaultRateLimit` or `GetMaxDefaultEndpointRateLimit`.

Again to keep the function names precise.

It's not a max rate limit for endpoints in the domain; it's max for the default rate limit for endpoints in the domain.

The JSON API doesn't convery this nuance well, but we can in the Go SDK.

Reply:
That's a good point. I was trying to model the API with my function names, but the Go SDK can allow us to explain some of the ambiguity in the API. I'll change the names to be more precise.